src/message.rs

```diff
@@ -101,14 +102,27 @@ pub trait Message: Debug + Send + Sync {
     ///
     /// The entire buffer will be consumed.
     fn merge<B>(&mut self, buf: B) -> Result<(), DecodeError>
+    where
+        B: IntoBuf,
+        Self: Sized,
+    {
+        self.merge_with_context(buf, DecodeContext::default())
+    }
+
+    /// Decodes an instance of the message from a buffer, and merges it into `self`.
+    /// This version takes a `DecodeContext` which contains some state for the
+    /// current decoding. Most users will want to use `merge` without the context.
+    ///
+    /// The entire buffer will be consumed.
+    fn merge_with_context<B>(&mut self, buf: B, ctx: DecodeContext) -> Result<(), DecodeError>
```

Comment:
AFAICT the only caller of this is the merge function right above -- could this be removed?

Reply:
If not, it should be marked doc hidden, since DecodeContext isn't part of the public API.

Reply:
Hmm, it turns out it is, I thought I called it from the code generated by prost-derive, but I don't